readme.md

```diff
@@ -0,0 +1,175 @@
+```yaml
+    # canopen_vcan is a configuration and is a configuration example for the CanOpen protocol.
+    # With a similar pattern you can create multiple canopen configurations.
+    # For now we use canopen_vcan as example. All these paramters inside canopen_vcan 
+    # are used for CargoDash's CanOpen implementation.
+    canopen_vcan:
+        
+        # Sets the bustype of the CanOpen network. 
+        bustype: socketcan 
+        
+        # Sets the channel name of the CanOpen network
+        channel: vcan0 
+        
+        # Sets the bitrate of CanOpen the network
+        bitrate: 500000 
+        
+        # Sets the nodes which are existing inside the network.
+        # This is done in an array with object that contain information needed for CargoDash.
+        # Every object needs to contain these attributes
+        #       local : sets if the node is either remote or local
+        #       eds_location : sets the location of the needed eds file of the node
+        #       node_purpose : object containing name and type of node:
+        #                      name: Short description of the node
+        #                      type: integer that should reflect the types of the nodes 
+        #                            at node_input_factory/node_input_enums.
+        #       For example the node_purpose type below is 0 and that is reflected with a
+        #       DistanceNode.
+
+        nodes: [{local: false, eds_location: 'eds_files/Arduino1.eds', node_purpose: {name: 'Front view object distance', type: 0}}]
+        
+        # Sets the max speed of wheelNodes.
+        max_speed_in_ms: 3 
+        
+        # Object related to can data logging.
+        # Our logger makes use of a buffered logger implementation.
+
+        raw_can_data_logging: {
+            # Enables of Disables the buffered logger.
+            enabled: true,
+
+            # The size of the buffer so with it can contain 1024 can messages.
+            buffer: 1024
+        }
+```
+## Extending CargoDash Communication Possibilities
+
+For now, CargoDash is only able to work with the CanOpen protocol but it has the potential to work with more communication protocols. If you want to add a communication protocol then you should add a new interpreter and a new listener. You can easily create a new listener class that uses the same methods as the abstract listener class. The same goes for the interpreter. It's important that you create nodes from the NodeFactory inside your interpreter. The NodeFactory class creates 'Node Input' objects which can be used to change the vehicle's state. In the diagram below you can have an understanding of CargoDash internal class interaction works.
+
+
+![CargoDash Architecture](img/api_cargodash_v6.png "CargoDash Architecture")
```

Comment:
dead link

Reply:
Volgens mij niet hoor, hier https://github.com/ZeroPoint095/CargoDash/tree/feature/CargoDashDocumentation  zie je dat de afbeelding wordt getoond

Reply:
ah, in the rich view van de PR niet. My bad